api/layer/object.go

```diff
@@ -145,6 +147,42 @@ func MimeByFileName(name string) string {
 	return mime.TypeByExtension(ext)
 }
 
+func encryptionReader(r io.Reader, size uint64, key []byte) (io.Reader, uint64, error) {
+	encSize, err := sio.EncryptedSize(size)
+	if err != nil {
+		return nil, 0, fmt.Errorf("failed to compute enc size: %w", err)
+	}
+
+	r, err = sio.EncryptReader(r, sio.Config{MinVersion: sio.Version20, MaxVersion: sio.Version20, Key: key, CipherSuites: []byte{sio.AES_256_GCM}})
+	if err != nil {
+		return nil, 0, fmt.Errorf("couldn't create encrypter: %w", err)
+	}
+
+	return r, encSize, nil
+}
+
+func parseCompletedPartHeader(hdr string) (*Part, error) {
+	// partInfo[0] -- part number, partInfo[1] -- part size, partInfo[2] -- checksum
```

Comment:
I'm not sure that it looks good that we have two copied code chunks. Can we call `parseCompletedPartHeader`  in `formUploadAttributes` and then cast `layer.Part` to `handler.Part`?